Formula/llvm.rb

```diff
@@ -211,7 +237,18 @@ def install
 
       if OS.mac?
         extra_args << "-DLLVM_ENABLE_LIBCXX=ON"
-        extra_args << "-DDEFAULT_SYSROOT=#{macos_sdk}" if macos_sdk
+        # Prevent CMake from defaulting to `lld` when it's found next to `clang`.
+        # This can be removed after CMake 3.25. See:
+        # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7671
+        extra_args << "-DLLVM_USE_LINKER=ld"
+        extra_args << "-DCMAKE_LINKER=ld"
+        extra_args += clt_sdk_support_flags
+
+        # NOTE: do not enable LTO on Linux, because this creates static archives that are not portable.
+        #       This is not an issue on macOS, where bottles are built and installed on the same version.
+        args << "-DLLVM_ENABLE_LTO=ON"
+        # LTO creates object files not recognised by Apple libtool.
+        args << "-DCMAKE_LIBTOOL=#{llvmpath}/stage1/bin/llvm-libtool-darwin"
```

Comment:
It seems that the supported way of doing this is to pass `DYLD_LIBRARY_PATH` to CMake:

https://reviews.llvm.org/D61346
https://github.com/llvm/llvm-project/blob/29c841ce93e087fa4e0c5f3abae94edd460bc24a/llvm/cmake/modules/UseLibtool.cmake#L39-L50

To my surprise, this does seem to work. That said, `llvm-libtool-darwin` has been in use in Compiler-RT since LLVM 12:

https://github.com/llvm/llvm-project/commit/2c80e2fe51b685f5663ad6f76c9c05c75d7582c4

So using `llvm-libtool-darwin` should be perfectly fine here too.